beacon-chain/core/helpers/rewards_penalties_test.go

```diff
@@ -184,6 +184,70 @@ func TestIncreaseBalance_OK(t *testing.T) {
 	}
 }
 
+func TestIncreaseBalanceAndAdjustPrincipalBalance(t *testing.T) {
+	// Define the test cases
+	tests := []struct {
+		name                     string
+		initialEffectiveBalance  uint64
+		initialPrincipalBalance  uint64
+		initialBalance           uint64
+		delta                    uint64
+		expectedPrincipalBalance uint64
+		expectedBalance          uint64
+	}{
+		{
+			name:                     "balance < principal balance < balance + delta",
+			initialEffectiveBalance:  4,
+			initialPrincipalBalance:  4,
+			initialBalance:           10,
+			delta:                    6,
+			expectedPrincipalBalance: 10,
+			expectedBalance:          16,
+		},
+		{
+			name:                     "balance < balance + delta < principal balance",
+			initialEffectiveBalance:  4,
+			initialPrincipalBalance:  14,
+			initialBalance:           10,
+			delta:                    6,
+			expectedPrincipalBalance: 16,
+			expectedBalance:          16,
+		},
```

Comment:
I’ve added the test cases.

```suggestion
		{
			name:                     "principal balance < balance",
			initialEffectiveBalance:  10,
			initialPrincipalBalance:  8,
			initialBalance:           10,
			delta:                    6,
			expectedPrincipalBalance: 14,
			expectedBalance:          16,
		},
		{
			name:                     "balance < principal balance < balance + delta",
			initialEffectiveBalance:  8,
			initialPrincipalBalance:  10,
			initialBalance:           8,
			delta:                    6,
			expectedPrincipalBalance: 14,
			expectedBalance:          14,
		},
		{
			name:                     "balance + delta < principal balance",
			initialEffectiveBalance:  10,
			initialPrincipalBalance:  10,
			initialBalance:           3,
			delta:                    6,
			expectedPrincipalBalance: 10,
			expectedBalance:          9,
		},
	}
```